docs/overlay.md

```diff
@@ -0,0 +1,21 @@
+---
+title: Overlay
+---
+
+- **Peer**
+    - A peer is another node in the Stellar network with which Stellar Core can communicate.
+- **Overlay**
+    - The overlay is a peer-to-peer network responsible for connecting to other Stellar cores (peers) and propagating transactions, blocks, and consensus votes to them.
+
+
+- **Overlay Network**
+    * Discovering and connecting to other peers on the network. The overlay has a built-in peer discovery mechanism, where peers exchnage known peers with each other over time. The initial set of connections to try is seeded via node's quorum set as well as `KNOWN_PEERS` and `PREFERRED_PEERS`. Connectivity targets are configured via `TARGET_PEER_CONNECTIONS` (max outbound) and `MAX_ADDITIONAL_PEER_CONNECTIONS` (max inbound). Core can expect up to `TARGET_PEER_CONNECTIONS`+`MAX_ADDITIONAL_PEER_CONNECTIONS` total authenticated connections.
+
+    * Connecting to peers. By default, overlay will select random peers it knows about to connect to. Connection preferences can be tailored via `KNOWN_PEERS` and `PREFERRED_PEERS` configs. The former is a list of known peers core will try to connect to without discovering them from others first. The latter is a list of peers to prioritize connecting to. If `PREFERRED_PEERS` or `PREFERRED_PEERS_KEYS` is configred, core will constatly try to connect to those peers.
```

Comment:
Small nit: the start of this bullet, the previous bullet, and the next bullet are incomplete sentences. It took me a second to realize that they're actually supposed to be subheaders. It might help to add something to visually distinguish them from the rest of the paragraph, such as a colon (like you do with the 'Versioning' bullet).